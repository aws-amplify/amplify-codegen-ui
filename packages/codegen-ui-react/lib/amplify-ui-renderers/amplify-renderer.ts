/*
  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.

  Licensed under the Apache License, Version 2.0 (the "License").
  You may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */
import { StudioNode, StudioComponent, StudioComponentChild } from '@aws-amplify/codegen-ui';
import { JsxElement, JsxFragment, JsxSelfClosingElement } from 'typescript';
// add primitives in alphabetical order
import {
  AlertProps,
  BadgeProps,
  ButtonProps,
  ButtonGroupProps,
  CardProps,
  CheckboxFieldProps,
  DividerProps,
  ExpanderProps,
  ExpanderItemProps,
  FlexProps,
  GridProps,
  HeadingProps,
  IconProps,
  ImageProps,
  LinkProps,
  LoaderProps,
  MenuItemProps,
  MenuProps,
  PaginationProps,
  PasswordFieldProps,
  PhoneNumberFieldProps,
  PlaceholderProps,
  RadioProps,
  RadioGroupFieldProps,
  RatingProps,
  ScrollViewProps,
  SearchFieldProps,
  SelectFieldProps,
  SliderFieldProps,
  StepperFieldProps,
  SwitchFieldProps,
  TabItemProps,
  TabsProps,
  TableCellProps,
  TableBodyProps,
  TableFootProps,
  TableHeadProps,
  TableProps,
  TableRowProps,
  TextAreaFieldProps,
  TextFieldProps,
  ToggleButtonProps,
  ToggleButtonGroupProps,
  ViewProps,
  VisuallyHiddenProps,
  TextProps,
} from '@aws-amplify/ui-react';
import { Primitive } from '../primitive';
import { ReactStudioTemplateRenderer } from '../react-studio-template-renderer';
import CustomComponentRenderer from './customComponent';
import CollectionRenderer from './collection';
import { ReactComponentRenderer } from '../react-component-renderer';

export class AmplifyRenderer extends ReactStudioTemplateRenderer {
  renderJsx(
    component: StudioComponent | StudioComponentChild,
    parent?: StudioNode,
  ): JsxElement | JsxFragment | JsxSelfClosingElement {
    const node = new StudioNode(component, parent);
    const renderChildren = (children: StudioComponentChild[]) => children.map((child) => this.renderJsx(child, node));

    // add Primitive in alphabetical order
    switch (component.componentType) {
      case Primitive.Alert:
        return new ReactComponentRenderer<AlertProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Badge:
        return new ReactComponentRenderer<BadgeProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Button:
        return new ReactComponentRenderer<ButtonProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.ButtonGroup:
        return new ReactComponentRenderer<ButtonGroupProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Card:
        return new ReactComponentRenderer<CardProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.CheckboxField:
        return new ReactComponentRenderer<CheckboxFieldProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Collection:
        return new CollectionRenderer(
          component,
          this.componentMetadata,
          this.importCollection,
          this.renderConfig,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Divider:
        return new ReactComponentRenderer<DividerProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement();

      case Primitive.Expander:
        return new ReactComponentRenderer<ExpanderProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.ExpanderItem:
        return new ReactComponentRenderer<ExpanderItemProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Flex:
        return new ReactComponentRenderer<FlexProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Grid:
        return new ReactComponentRenderer<GridProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Heading:
        return new ReactComponentRenderer<HeadingProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Icon:
        return new ReactComponentRenderer<IconProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Image:
        return new ReactComponentRenderer<ImageProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement();

      case Primitive.Link:
        return new ReactComponentRenderer<LinkProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Loader:
        return new ReactComponentRenderer<LoaderProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.MenuButton:
        return new ReactComponentRenderer<ButtonProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.MenuItem:
        return new ReactComponentRenderer<MenuItemProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Menu:
        return new ReactComponentRenderer<MenuProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Pagination:
        return new ReactComponentRenderer<PaginationProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.PasswordField:
        return new ReactComponentRenderer<PasswordFieldProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.PhoneNumberField:
        return new ReactComponentRenderer<PhoneNumberFieldProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Placeholder:
        return new ReactComponentRenderer<PlaceholderProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Radio:
        return new ReactComponentRenderer<RadioProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.RadioGroupField:
        return new ReactComponentRenderer<RadioGroupFieldProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Rating:
        return new ReactComponentRenderer<RatingProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.ScrollView:
        return new ReactComponentRenderer<ScrollViewProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.SearchField:
        return new ReactComponentRenderer<SearchFieldProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.SelectField:
        return new ReactComponentRenderer<SelectFieldProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.SliderField:
        return new ReactComponentRenderer<SliderFieldProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement();

      case Primitive.StepperField:
        return new ReactComponentRenderer<StepperFieldProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.SwitchField:
        return new ReactComponentRenderer<SwitchFieldProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.TabItem:
        return new ReactComponentRenderer<TabItemProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Tabs:
        return new ReactComponentRenderer<TabsProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Table:
        return new ReactComponentRenderer<TableProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.TableBody:
        return new ReactComponentRenderer<TableBodyProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.TableCell:
        return new ReactComponentRenderer<TableCellProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.TableFoot:
        return new ReactComponentRenderer<TableFootProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.TableHead:
        return new ReactComponentRenderer<TableHeadProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.TableRow:
        return new ReactComponentRenderer<TableRowProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.Text:
        return new ReactComponentRenderer<TextProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.TextAreaField:
        return new ReactComponentRenderer<TextAreaFieldProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.TextField:
        return new ReactComponentRenderer<TextFieldProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.ToggleButton:
        return new ReactComponentRenderer<ToggleButtonProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.ToggleButtonGroup:
        return new ReactComponentRenderer<ToggleButtonGroupProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.View:
        return new ReactComponentRenderer<ViewProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      case Primitive.VisuallyHidden:
        return new ReactComponentRenderer<VisuallyHiddenProps>(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);

      default:
        return new CustomComponentRenderer(
          component,
          this.componentMetadata,
          this.importCollection,
          parent,
        ).renderElement(renderChildren);
    }
  }
}
