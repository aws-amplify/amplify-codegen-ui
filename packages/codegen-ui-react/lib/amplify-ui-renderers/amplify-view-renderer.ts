/*
  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.

  Licensed under the Apache License, Version 2.0 (the "License").
  You may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */

import { JsxElement, factory, JsxFragment } from 'typescript';
import { ReactViewTemplateRenderer } from '../views/react-view-renderer';
import { Primitive } from '../primitive';
import { ReactTableRenderer } from '../react-table-renderer';

export class AmplifyViewRenderer extends ReactViewTemplateRenderer {
  renderJsx(): JsxElement | JsxFragment {
    switch (this.viewComponent.viewConfiguration.type) {
      case Primitive.Table:
        return new ReactTableRenderer(
          this.viewComponent,
          this.viewDefinition,
          this.viewMetadata,
          this.importCollection,
        ).renderElement();
      default:
        return factory.createJsxFragment(factory.createJsxOpeningFragment(), [], factory.createJsxJsxClosingFragment());
    }
  }
}
