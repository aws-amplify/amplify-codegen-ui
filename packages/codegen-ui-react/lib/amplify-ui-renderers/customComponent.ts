/*
  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.

  Licensed under the Apache License, Version 2.0 (the "License").
  You may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */
import { StudioComponentChild } from '@aws-amplify/codegen-ui';
import { JsxChild, JsxElement, factory } from 'typescript';
import { isAliased, removeAlias } from '../helpers';
import { ReactComponentRenderer } from '../react-component-renderer';

export default class CustomComponentRenderer<TPropIn> extends ReactComponentRenderer<TPropIn> {
  renderElement(renderChildren: (children: StudioComponentChild[]) => JsxChild[]): JsxElement {
    const children = this.component.children ? this.component.children : [];
    const element = factory.createJsxElement(
      this.renderOpeningElement(),
      renderChildren(children),
      factory.createJsxClosingElement(factory.createIdentifier(this.component.componentType)),
    );

    if (isAliased(this.component.componentType)) {
      this.importCollection.addImport(
        `./${removeAlias(this.component.componentType)}`,
        `${removeAlias(this.component.componentType)} as ${this.component.componentType}`,
      );
    } else {
      this.importCollection.addImport(`./${this.component.componentType}`, 'default');
    }

    return element;
  }
}
