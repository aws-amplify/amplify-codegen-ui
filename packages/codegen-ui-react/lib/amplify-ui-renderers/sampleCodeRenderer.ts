/*
  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.

  Licensed under the Apache License, Version 2.0 (the "License").
  You may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */
import { BaseComponentProps } from '@aws-amplify/ui-react';

import { factory, JsxAttribute, JsxAttributeLike, JsxSelfClosingElement } from 'typescript';
import { FrontendManagerRendererConstants } from '@aws-amplify/codegen-ui';
import { ReactComponentRenderer } from '../react-component-renderer';

export default class SampleCodeRenderer extends ReactComponentRenderer<BaseComponentProps> {
  renderElement(): JsxSelfClosingElement {
    const tagName = this.component.name ?? FrontendManagerRendererConstants.unknownName;
    // const prop = new Map<string, BoundFrontendManagerComponentProperty>();
    // this.collectExposedProps(this.component, prop);

    const propsArray: JsxAttribute[] = [];
    /*  TODO:  move over to boundProperties
    const defaultValue = 'defaultValue';
    const defaultValueExpr = factory.createJsxExpression(undefined, factory.createIdentifier(defaultValue));
    props?.forEach((value, key) => {
      if (value.exposedAs) {
        const displayExpr =
          value.value !== undefined ? factory.createStringLiteral(value.value.toString()) : defaultValueExpr;
        const attr = factory.createJsxAttribute(factory.createIdentifier(key), displayExpr);
        propsArray.push(attr);
      }
    });
    */

    return factory.createJsxSelfClosingElement(
      factory.createIdentifier(tagName),
      undefined,
      factory.createJsxAttributes(propsArray),
    );
  }

  /*
  private collectExposedProps(
    component: FrontendManagerComponent,
    collected: Map<string, FrontendManagerComponentProperty>,
  ) {
    const moreItems = Object.entries(component.properties).filter((m) => !(m[1].exposedAs == null));
    moreItems?.forEach((value, index) => {
      if (!collected.has(value[0])) {
        collected.set(value[0], value[1]);
      }
    });

    component.children?.forEach((child) => {
      this.collectExposedProps(child, collected);
    });
  }
  */

  private addExposedPropAttributes(attributes: JsxAttributeLike[], tagName: string) {
    const propsAttr = factory.createJsxSpreadAttribute(factory.createIdentifier('props'));
    attributes.push(propsAttr);

    const overrideAttr = factory.createJsxSpreadAttribute(
      factory.createCallExpression(factory.createIdentifier('getOverrideProps'), undefined, [
        factory.createPropertyAccessExpression(
          factory.createIdentifier('props'),
          factory.createIdentifier('overrides'),
        ),
        factory.createStringLiteral(tagName),
      ]),
    );
    attributes.push(overrideAttr);
  }

  mapProps(props: BaseComponentProps): BaseComponentProps {
    return props;
  }
}
