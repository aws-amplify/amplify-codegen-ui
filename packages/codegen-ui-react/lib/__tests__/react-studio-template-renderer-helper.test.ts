/*
  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.

  Licensed under the Apache License, Version 2.0 (the "License").
  You may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */
import { StudioTemplateRendererFactory, StudioComponent } from '@aws-amplify/codegen-ui';
import { ScriptTarget, ScriptKind } from '..';
import { AmplifyRenderer } from '../amplify-ui-renderers/amplify-renderer';
import { formatCode } from '../react-studio-template-renderer-helper';
import { loadSchemaFromJSONFile } from './__utils__';

function generateDeclarationForScriptTarget(jsonSchemaFile: string, target: ScriptTarget): any {
  const rendererFactory = new StudioTemplateRendererFactory(
    (component: StudioComponent) =>
      new AmplifyRenderer(component, { target, script: ScriptKind.JS, renderTypeDeclarations: true }),
  );
  const component = rendererFactory.buildRenderer(loadSchemaFromJSONFile(jsonSchemaFile)).renderComponent();
  return (component as unknown as { declaration: string }).declaration;
}

describe('react-studio-template-renderer-helper', () => {
  describe('transpile', () => {
    it('successfully transpiles with ScriptTarget ES3', () => {
      expect(generateDeclarationForScriptTarget('custom/customParentAndChildren', ScriptTarget.ES3)).toMatchSnapshot();
    });

    it('successfully transpiles with ScriptTarget ES5', () => {
      expect(generateDeclarationForScriptTarget('custom/customParentAndChildren', ScriptTarget.ES5)).toMatchSnapshot();
    });

    it('successfully transpiles with ScriptTarget ES2015', () => {
      expect(
        generateDeclarationForScriptTarget('custom/customParentAndChildren', ScriptTarget.ES2015),
      ).toMatchSnapshot();
    });

    it('successfully transpiles with ScriptTarget ES2016', () => {
      expect(
        generateDeclarationForScriptTarget('custom/customParentAndChildren', ScriptTarget.ES2016),
      ).toMatchSnapshot();
    });

    it('successfully transpiles with ScriptTarget ES2017', () => {
      expect(
        generateDeclarationForScriptTarget('custom/customParentAndChildren', ScriptTarget.ES2017),
      ).toMatchSnapshot();
    });

    it('successfully transpiles with ScriptTarget ES2018', () => {
      expect(
        generateDeclarationForScriptTarget('custom/customParentAndChildren', ScriptTarget.ES2018),
      ).toMatchSnapshot();
    });

    it('successfully transpiles with ScriptTarget ES2019', () => {
      expect(
        generateDeclarationForScriptTarget('custom/customParentAndChildren', ScriptTarget.ES2019),
      ).toMatchSnapshot();
    });

    it('successfully transpiles with ScriptTarget ES2020', () => {
      expect(
        generateDeclarationForScriptTarget('custom/customParentAndChildren', ScriptTarget.ES2020),
      ).toMatchSnapshot();
    });

    it('successfully transpiles with ScriptTarget ES2021', () => {
      expect(
        generateDeclarationForScriptTarget('custom/customParentAndChildren', ScriptTarget.ES2021),
      ).toMatchSnapshot();
    });

    it('fails to transpile with ScriptTarget ESNext', () => {
      expect(() => {
        generateDeclarationForScriptTarget('custom/customParentAndChildren', ScriptTarget.ESNext);
      }).toThrowErrorMatchingSnapshot();
    });

    it('fails to transpile with ScriptTarget Latest', () => {
      expect(() => {
        generateDeclarationForScriptTarget('custom/customParentAndChildren', ScriptTarget.Latest);
      }).toThrowErrorMatchingSnapshot();
    });
  });

  describe('formatCode', () => {
    const code = 'const foo = 1;     const bar    =  false;';

    it('formats code when prettier is installed', () => {
      expect(formatCode(code)).toMatchSnapshot();
    });

    it('does not format code when prettier is not installed', () => {
      jest.mock('prettier', () => {
        throw new Error('Mock module not installed.');
      });

      expect(formatCode(code)).toEqual(code);
    });

    it('does not swallow prettier errors', () => {
      jest.mock('prettier', () => ({
        format: jest.fn(() => {
          throw new Error('Failed to format');
        }),
      }));

      expect(() => formatCode(code)).toThrowErrorMatchingSnapshot();
    });
  });
});
