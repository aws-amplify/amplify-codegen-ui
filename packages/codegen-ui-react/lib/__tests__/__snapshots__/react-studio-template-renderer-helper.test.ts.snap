// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-studio-template-renderer-helper formatCode does not swallow prettier errors 1`] = `"Failed to format"`;

exports[`react-studio-template-renderer-helper formatCode formats code when prettier is installed 1`] = `
"const foo = 1;
const bar = false;
"
`;

exports[`react-studio-template-renderer-helper transpile fails to transpile with ScriptTarget ESNext 1`] = `"ScriptTarget 99 not supported with type declarations enabled, expected one of [0,1,2,3,4,5,6,7,8]"`;

exports[`react-studio-template-renderer-helper transpile fails to transpile with ScriptTarget Latest 1`] = `"ScriptTarget 99 not supported with type declarations enabled, expected one of [0,1,2,3,4,5,6,7,8]"`;

exports[`react-studio-template-renderer-helper transpile successfully transpiles with ScriptTarget ES3 1`] = `
"import * as React from \\"react\\";
import { CustomButtonProps } from \\"./CustomButton\\";
import { ViewTestProps } from \\"./ViewTest\\";
export declare type EscapeHatchProps = {
    [elementHierarchy: string]: Record<string, unknown>;
} | null;
export declare type VariantValues = {
    [key: string]: string;
};
export declare type Variant = {
    variantValues: VariantValues;
    overrides: EscapeHatchProps;
};
export declare type PrimitiveOverrideProps<T> = Partial<T> & React.DOMAttributes<HTMLDivElement>;
export declare type CustomParentAndChildrenOverridesProps = {
    CustomParentAndChildren?: ViewTestProps;
    MyCustomButton?: CustomButtonProps;
} & EscapeHatchProps;
export declare type CustomParentAndChildrenProps = React.PropsWithChildren<Partial<ViewTestProps> & {
    overrides?: CustomParentAndChildrenOverridesProps | undefined | null;
}>;
export default function CustomParentAndChildren(props: CustomParentAndChildrenProps): React.ReactElement;
"
`;

exports[`react-studio-template-renderer-helper transpile successfully transpiles with ScriptTarget ES5 1`] = `
"import * as React from \\"react\\";
import { CustomButtonProps } from \\"./CustomButton\\";
import { ViewTestProps } from \\"./ViewTest\\";
export declare type EscapeHatchProps = {
    [elementHierarchy: string]: Record<string, unknown>;
} | null;
export declare type VariantValues = {
    [key: string]: string;
};
export declare type Variant = {
    variantValues: VariantValues;
    overrides: EscapeHatchProps;
};
export declare type PrimitiveOverrideProps<T> = Partial<T> & React.DOMAttributes<HTMLDivElement>;
export declare type CustomParentAndChildrenOverridesProps = {
    CustomParentAndChildren?: ViewTestProps;
    MyCustomButton?: CustomButtonProps;
} & EscapeHatchProps;
export declare type CustomParentAndChildrenProps = React.PropsWithChildren<Partial<ViewTestProps> & {
    overrides?: CustomParentAndChildrenOverridesProps | undefined | null;
}>;
export default function CustomParentAndChildren(props: CustomParentAndChildrenProps): React.ReactElement;
"
`;

exports[`react-studio-template-renderer-helper transpile successfully transpiles with ScriptTarget ES2015 1`] = `
"import * as React from \\"react\\";
import { CustomButtonProps } from \\"./CustomButton\\";
import { ViewTestProps } from \\"./ViewTest\\";
export declare type EscapeHatchProps = {
    [elementHierarchy: string]: Record<string, unknown>;
} | null;
export declare type VariantValues = {
    [key: string]: string;
};
export declare type Variant = {
    variantValues: VariantValues;
    overrides: EscapeHatchProps;
};
export declare type PrimitiveOverrideProps<T> = Partial<T> & React.DOMAttributes<HTMLDivElement>;
export declare type CustomParentAndChildrenOverridesProps = {
    CustomParentAndChildren?: ViewTestProps;
    MyCustomButton?: CustomButtonProps;
} & EscapeHatchProps;
export declare type CustomParentAndChildrenProps = React.PropsWithChildren<Partial<ViewTestProps> & {
    overrides?: CustomParentAndChildrenOverridesProps | undefined | null;
}>;
export default function CustomParentAndChildren(props: CustomParentAndChildrenProps): React.ReactElement;
"
`;

exports[`react-studio-template-renderer-helper transpile successfully transpiles with ScriptTarget ES2016 1`] = `
"import * as React from \\"react\\";
import { CustomButtonProps } from \\"./CustomButton\\";
import { ViewTestProps } from \\"./ViewTest\\";
export declare type EscapeHatchProps = {
    [elementHierarchy: string]: Record<string, unknown>;
} | null;
export declare type VariantValues = {
    [key: string]: string;
};
export declare type Variant = {
    variantValues: VariantValues;
    overrides: EscapeHatchProps;
};
export declare type PrimitiveOverrideProps<T> = Partial<T> & React.DOMAttributes<HTMLDivElement>;
export declare type CustomParentAndChildrenOverridesProps = {
    CustomParentAndChildren?: ViewTestProps;
    MyCustomButton?: CustomButtonProps;
} & EscapeHatchProps;
export declare type CustomParentAndChildrenProps = React.PropsWithChildren<Partial<ViewTestProps> & {
    overrides?: CustomParentAndChildrenOverridesProps | undefined | null;
}>;
export default function CustomParentAndChildren(props: CustomParentAndChildrenProps): React.ReactElement;
"
`;

exports[`react-studio-template-renderer-helper transpile successfully transpiles with ScriptTarget ES2017 1`] = `
"import * as React from \\"react\\";
import { CustomButtonProps } from \\"./CustomButton\\";
import { ViewTestProps } from \\"./ViewTest\\";
export declare type EscapeHatchProps = {
    [elementHierarchy: string]: Record<string, unknown>;
} | null;
export declare type VariantValues = {
    [key: string]: string;
};
export declare type Variant = {
    variantValues: VariantValues;
    overrides: EscapeHatchProps;
};
export declare type PrimitiveOverrideProps<T> = Partial<T> & React.DOMAttributes<HTMLDivElement>;
export declare type CustomParentAndChildrenOverridesProps = {
    CustomParentAndChildren?: ViewTestProps;
    MyCustomButton?: CustomButtonProps;
} & EscapeHatchProps;
export declare type CustomParentAndChildrenProps = React.PropsWithChildren<Partial<ViewTestProps> & {
    overrides?: CustomParentAndChildrenOverridesProps | undefined | null;
}>;
export default function CustomParentAndChildren(props: CustomParentAndChildrenProps): React.ReactElement;
"
`;

exports[`react-studio-template-renderer-helper transpile successfully transpiles with ScriptTarget ES2018 1`] = `
"import * as React from \\"react\\";
import { CustomButtonProps } from \\"./CustomButton\\";
import { ViewTestProps } from \\"./ViewTest\\";
export declare type EscapeHatchProps = {
    [elementHierarchy: string]: Record<string, unknown>;
} | null;
export declare type VariantValues = {
    [key: string]: string;
};
export declare type Variant = {
    variantValues: VariantValues;
    overrides: EscapeHatchProps;
};
export declare type PrimitiveOverrideProps<T> = Partial<T> & React.DOMAttributes<HTMLDivElement>;
export declare type CustomParentAndChildrenOverridesProps = {
    CustomParentAndChildren?: ViewTestProps;
    MyCustomButton?: CustomButtonProps;
} & EscapeHatchProps;
export declare type CustomParentAndChildrenProps = React.PropsWithChildren<Partial<ViewTestProps> & {
    overrides?: CustomParentAndChildrenOverridesProps | undefined | null;
}>;
export default function CustomParentAndChildren(props: CustomParentAndChildrenProps): React.ReactElement;
"
`;

exports[`react-studio-template-renderer-helper transpile successfully transpiles with ScriptTarget ES2019 1`] = `
"import * as React from \\"react\\";
import { CustomButtonProps } from \\"./CustomButton\\";
import { ViewTestProps } from \\"./ViewTest\\";
export declare type EscapeHatchProps = {
    [elementHierarchy: string]: Record<string, unknown>;
} | null;
export declare type VariantValues = {
    [key: string]: string;
};
export declare type Variant = {
    variantValues: VariantValues;
    overrides: EscapeHatchProps;
};
export declare type PrimitiveOverrideProps<T> = Partial<T> & React.DOMAttributes<HTMLDivElement>;
export declare type CustomParentAndChildrenOverridesProps = {
    CustomParentAndChildren?: ViewTestProps;
    MyCustomButton?: CustomButtonProps;
} & EscapeHatchProps;
export declare type CustomParentAndChildrenProps = React.PropsWithChildren<Partial<ViewTestProps> & {
    overrides?: CustomParentAndChildrenOverridesProps | undefined | null;
}>;
export default function CustomParentAndChildren(props: CustomParentAndChildrenProps): React.ReactElement;
"
`;

exports[`react-studio-template-renderer-helper transpile successfully transpiles with ScriptTarget ES2020 1`] = `
"import * as React from \\"react\\";
import { CustomButtonProps } from \\"./CustomButton\\";
import { ViewTestProps } from \\"./ViewTest\\";
export declare type EscapeHatchProps = {
    [elementHierarchy: string]: Record<string, unknown>;
} | null;
export declare type VariantValues = {
    [key: string]: string;
};
export declare type Variant = {
    variantValues: VariantValues;
    overrides: EscapeHatchProps;
};
export declare type PrimitiveOverrideProps<T> = Partial<T> & React.DOMAttributes<HTMLDivElement>;
export declare type CustomParentAndChildrenOverridesProps = {
    CustomParentAndChildren?: ViewTestProps;
    MyCustomButton?: CustomButtonProps;
} & EscapeHatchProps;
export declare type CustomParentAndChildrenProps = React.PropsWithChildren<Partial<ViewTestProps> & {
    overrides?: CustomParentAndChildrenOverridesProps | undefined | null;
}>;
export default function CustomParentAndChildren(props: CustomParentAndChildrenProps): React.ReactElement;
"
`;

exports[`react-studio-template-renderer-helper transpile successfully transpiles with ScriptTarget ES2021 1`] = `
"import * as React from \\"react\\";
import { CustomButtonProps } from \\"./CustomButton\\";
import { ViewTestProps } from \\"./ViewTest\\";
export declare type EscapeHatchProps = {
    [elementHierarchy: string]: Record<string, unknown>;
} | null;
export declare type VariantValues = {
    [key: string]: string;
};
export declare type Variant = {
    variantValues: VariantValues;
    overrides: EscapeHatchProps;
};
export declare type PrimitiveOverrideProps<T> = Partial<T> & React.DOMAttributes<HTMLDivElement>;
export declare type CustomParentAndChildrenOverridesProps = {
    CustomParentAndChildren?: ViewTestProps;
    MyCustomButton?: CustomButtonProps;
} & EscapeHatchProps;
export declare type CustomParentAndChildrenProps = React.PropsWithChildren<Partial<ViewTestProps> & {
    overrides?: CustomParentAndChildrenOverridesProps | undefined | null;
}>;
export default function CustomParentAndChildren(props: CustomParentAndChildrenProps): React.ReactElement;
"
`;
